include/nbl/builtin/hlsl/prefix_sum_blur/box_sampler.hlsl

```diff
@@ -0,0 +1,183 @@
+                const T lastMinusOne = prefixSumAccessor.template get<T, uint32_t>(lastIdx - 1);
+                result += (rightIdx - lastIdx) * (last - lastMinusOne) + last;
+                break;
+            }
+            case ETC_MIRROR:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                T floored, ceiled;
+                int32_t d = rightFlIdx - lastIdx;
+
+                if (d % (2 * linearSize) == linearSize)
+                    floored = ((d + linearSize) / linearSize) * last;
+                else
+                {
+                    const uint32_t period = uint32_t(ceil(float32_t(d) / linearSize));
+                    if ((period & 0x1u) == 1)
+                        floored = period * last + last - prefixSumAccessor.template get<T, uint32_t>(lastIdx - uint32_t(d % linearSize));
+                    else
+                        floored = period * last + prefixSumAccessor.template get<T, uint32_t>((d - 1) % linearSize);
+                }
+                
+                d = rightClIdx - lastIdx;
+                if (d % (2 * linearSize) == linearSize)
+                    ceiled = ((d + linearSize) / linearSize) * last;
+                else
+                {
+                    const uint32_t period = uint32_t(ceil(float32_t(d) / linearSize));
+                    if ((period & 0x1u) == 1)
+                        ceiled = period * last + last - prefixSumAccessor.template get<T, uint32_t>(lastIdx - uint32_t(d % linearSize));
+                    else
+                        ceiled = period * last + prefixSumAccessor.template get<T, uint32_t>((d - 1) % linearSize);
+                }
+
+                result += lerp(floored, ceiled, alpha);
+                break;
+            }
+            case ETC_MIRROR_CLAMP_TO_EDGE:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T first = prefixSumAccessor.template get<T, uint32_t>(0);
+                result += (rightIdx - lastIdx) * first + last;
+                break;
+            }
+            }
+        }
+
+        if (leftFlIdx >= 0)
+        {
+            result -= lerp(prefixSumAccessor.template get<T, uint32_t>(leftFlIdx), prefixSumAccessor.template get<T, uint32_t>(leftClIdx), alpha);
+        }
+        else
+        {
+            switch (wrapMode) {
+            case ETC_REPEAT:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T floored = prefixSumAccessor.template get<T, uint32_t>((lastIdx + leftFlIdx) % linearSize) + floor(T(leftFlIdx) / linearSize) * last;
+                const T ceiled = prefixSumAccessor.template get<T, uint32_t>((lastIdx + leftClIdx) % linearSize) + floor(T(leftClIdx) / linearSize) * last;
+                result -= lerp(floored, ceiled, alpha);

```

Comment:
its interesting how you've allowed this to compute `last` multiple properly but not the right side case

doesn't the non symmetry of your cases not disturb you immediately?